runtime/compiler/runtime/Trampoline.cpp

```diff
@@ -101,7 +97,7 @@ void ppcCreateHelperTrampolines(uint8_t *trampPtr, int32_t numHelpers)
 
             // Now, if highest bit is on we need to clear the sign extend bits on 64bit CPUs
             // ** POWER4 pref fix **
-            if( (helper & 0x80000000) && (!customP4 || proc == TR_PPCgp))
+            if( (helper & 0x80000000) && (!customP4 || (customP4 && TR::comp()->target().cpu.is(OMR_PROCESSOR_PPC_GP))))
```

Comment:
The `customP4 &&` is not necessary, i'll get rid of it.

Reply:
better not to touch these portion of code.  i have a waiting commit to be merged for reworking trampolines etc for POWER10.

that customP4 is actually meaningful for POWER4 machines (there was a performance bug in hardware for countCache prediction ... that customP4 was to work around that bug).

Reply:
Are you suggesting to leave the change in as it is?

I'm turning it into
`if( (helper & 0x80000000) && (!customP4 || TR::comp()->target().cpu.is(OMR_PROCESSOR_PPC_GP)))`
which should be equivalent.

Reply:
I thought you are trying to throw away customP4 all together. In case, something wrong is unnecessarily committed.  Fine, that equiv transformation alone looks ok.